src/shell/mysh.c

```diff
@@ -1,7 +1,14 @@
 #include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+
+#define STDIN_FILENO 0
```

Comment:
unistd.h defines STDIN_FILENO, etc, see http://pubs.opengroup.org/onlinepubs/7908799/xsh/unistd.h.html, so I don't think you need these define statements